### ISSUE TITLE
namedtuple instances are not recognized as classes (anymore)

### ISSUE THREAD

**Issue by winfr34k** [PyCQA/pylint#1530]
### Steps to reproduce
1. Create a Python 3 file like so:
```python
"""
This is  a little test module.
"""
import collections


class MyClass:
    """
    Test class
    """
    def __init__(self, field1, field2):
        self.field1 = field1
        self.field2 = field2

    def do_something1(self):
        """
        Test method 1
        """
        pass

    def do_somethin2(self):
        """
        Test method 2
        """
        pass

    def __str__(self):
        return "MyClass(field1=%s,field2=%s)" % (self.field1, self.field2)


MyTuple = collections.namedtuple(typename="MyTuple",
                                 field_names=["field1", "field2"])


print(MyTuple(field1="test", field2="test2"))
print(MyClass(field1="test", field2="test2"))
``` 
2. Run pylint

**Note:** It doesn't matter if you write the import as `import collections` or `from collections import namedtuple`.

### Current behavior
Pylint **doesn't** recognize `MyTuple` as a `class` and therefore reports:
```
☁  test  pylint test_namedtuple2.py
No config file found, using default configuration
************* Module test_namedtuple2
C: 31, 0: Invalid constant name "MyTuple" (invalid-name)

------------------------------------------------------------------
Your code has been rated at 9.29/10 (previous run: 8.93/10, +0.36)
```
...as it's using the constants regex instead of the class regex.

### Expected behavior
Pylint recognizes `namedtuple` instances as classes  and uses the class regex instead of the constant regex, resulting in an output like so:
```
☁  test  pylint test_namedtuple1.py
No config file found, using default configuration

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### pylint --version output
```
☁  test  pylint --version
No config file found, using default configuration
pylint 1.7.1,
astroid 1.5.3
Python 3.4.2 (v3.4.2:ab2c023a9432, Oct  5 2014, 20:42:22)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

---

I have read in your changelogs that this once worked in version 1.0.0: https://docs.pylint.org/en/latest/whatsnew/changelog.html?highlight=namedtuple#what-s-new-in-pylint-1-0-0

**Comment by PCManticore** [comment]
No, the problem is that you are using keyword arguments for creating the namedtuple class, e.g ``namedtuple(typename=..., field_names=...)``. If instead you would write ``MyClass = namedtuple('MyClass', '...')``, this should work and invalid-name won't be emitted any longer.

**Comment by winfr34k** [comment]
That is really strange behaviour IMHO. Pylint should care about the result of the expression instead of how it's called... But good to know. I'll try it out tomorrow.

**Comment by PCManticore** [comment]
Yep, it is a bit weird, but that's because we have to infer the result of the expression. Without checking the parameters, we might infer something that is not a namedtuple per se. Anyway, this is fixed now, should work fine.
